systems/ps5.ixx

```diff
@@ -0,0 +1,151 @@
+module;
+#include <algorithm>
+#include <array>
+#include <filesystem>
+#include <format>
+#include <map>
+#include <ostream>
+#include "system.hh"
+#include "throw_line.hh"
+
+export module systems.ps5;
+
+import filesystem.iso9660;
+import readers.sector_reader;
+import utils.misc;
+import utils.strings;
+
+
+
+namespace gpsxre
+{
+
+export class SystemPS5 : public System
+{
+public:
+    std::string getName() override
+    {
+        return "PS5";
+    }
+
+
+    Type getType() override
+    {
+        return Type::ISO;
+    }
+
+
+    void printInfo(std::ostream &os, SectorReader *sector_reader, const std::filesystem::path &) const override
+    {
+        iso9660::PrimaryVolumeDescriptor pvd;
+        if(!iso9660::Browser::findDescriptor((iso9660::VolumeDescriptor &)pvd, sector_reader, iso9660::VolumeDescriptorType::PRIMARY))
+            return;
+        auto root_directory = iso9660::Browser::rootDirectory(sector_reader, pvd);
+
+        auto param_json = loadJSON(root_directory, "bd/param.json");
+
+        auto it = param_json.find("masterVersion");
+        if(it != param_json.end())
+            os << std::format("  version: {}", it->second) << std::endl;
+
+        it = param_json.find("masterDataId");
+        if(it != param_json.end())
+            os << std::format("  serial: {}", it->second) << std::endl;
+    }
+
+private:
+    std::map<std::string, std::string> loadJSON(std::shared_ptr<iso9660::Entry> root_directory, std::string json_file) const
+    {
+        std::map<std::string, std::string> json;
+
+        auto json_entry = root_directory->subEntry(json_file);
+        if(json_entry)
+        {
+            auto data = json_entry->read();
+            if(data.size() <= 0x800)
+                return json;
+
+            data.erase(data.begin(), data.begin() + 0x800);
+
+            // Parse JSON into key/value pairs
+            size_t cur = 0;
+            while(cur < data.size())
```

Comment:
nit: As you don't need cur outside of the loop, for() loop is preferable over while(), could do:
```
for(size_t cur = 0; cur < data.size(); )
```